src/lib.rs

```diff
@@ -831,6 +831,32 @@ pub unsafe trait KnownLayout {
         let meta = Self::PointerMetadata::from_elem_count(0);
         Self::raw_from_ptr_len(NonNull::dangling(), meta)
     }
+
+    /// Computes the size of an object of type `Self` with the given pointer
+    /// metadata.
+    ///
+    /// # Safety
+    ///
+    /// `size_for_metadata` promises to return `None` if and only if the
+    /// resulting size would not fit in a `usize`. Note that the returned size
+    /// could exceed the actual maximum valid size of an allocated object,
+    /// `isize::MAX`.
```

Comment:
Can you add an example that demonstrates this?

Reply:
Besides `assert_eq!(<[u16]>::size_for_metadata(usize::MAX), None);`?

Reply:
[Done](https://github.com/google/zerocopy/compare/353b5cbf72415fdedb5c1a9e05c65595e94876b1..bbbfc79988c84864de77b75a89d9dc0a6689430b#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R855-R856)